identity-verification/module_2/2.1 Rekognition IDV API Examples.ipynb

```diff
@@ -0,0 +1,706 @@
+    "    \n",
+    "The following steps are used to create a collection, index faces and finally search the collection for a face. \n",
+    "\n",
+    "1. [CreateCollection](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_CreateCollection.html\") Creates a collection in an AWS Region.\n",
+    "2. [IndexFaces](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_IndexFaces.html\")Detects faces in the input image and adds them to the specified collection.\n",
+    "3. [SearchFacesByImage](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_SearchFacesByImage.html\") For a given input image, first detects the largest face in the image, and then searches the specified collection for matching faces. The operation compares the features of the input face with faces in the specified collection.\n",
+    "4. [SearchFaces](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_SearchFacesByImage.html\") For a given input face ID, searches for matching faces in the collection the face belongs to. You get a face ID when you add a face to the collection using the IndexFaces operation. The operation compares the features of the input face with faces in the specified collection.\n",
+    "5. [DeleteFaces](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_DeleteFaces.html\")Deletes faces from a collection. You specify a collection ID and an array of face IDs to remove from the collection.\n",
+    "\n",
+    "\n",
+    "#### Other Collection APIs to try.\n",
+    "------\n",
+    "- [ListCollections](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_ListCollections.html\")Returns list of collection IDs in your account. \n",
+    "- [ListFaces](\"https://docs.aws.amazon.com/rekognition/latest/APIReference/API_ListFaces.html\")Returns metadata for faces in the specified collection. This metadata includes information such as the bounding box coordinates, the confidence (that the bounding box contains a face), and face ID.\n",
+    "\n",
+    "### Name your Collection \n",
+    "\n",
+    "------\n",
+    "\n",
+    "Here you can update the name of your collection\n",
+    "\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Name your Collection \n",
+    "collection_name = 'identity-verification-collection'\n",
+    "\n",
+    "try:\n",
+    "    # try and create a colleciton, \n",
+    "    response = rek_client.create_collection(\n",
+    "        CollectionId=collection_name)\n",
+    "except:\n",
+    "    # if it exists, delete it then created it \n",
+    "    response = rek_client.delete_collection(\n",
+    "        CollectionId=collection_name\n",
+    "    )\n",
+    "    response = rek_client.create_collection(\n",
+    "        CollectionId=collection_name\n",
+    "    )\n",
+    "\n",
+    "response\n"
```

Comment:
Have you thought about the implications of this if a customer already has an 'identity-verification-collection' collection in their account that isn't this demo?

Reply:
My interpretation of the comment `name your collection` is that we give them the opportunity to override. Are you suggesting we suggest an empty default?

Reply:
I don't have a suggested implementation. Just that if we're going to be deleting things, we want to be **very** sure we're not deleting something unrelated in the account. Expecting that someone trying out this demo will read and understand all of the code probably isn't good enough.. it might require changes either to the code or to the supporting documentation.

Reply:
I removed the collection name and added a comment to name their collection

Reply:
done